sos/report/plugins/block.py

```diff
@@ -56,7 +56,14 @@ def setup(self):
             "udevadm info %(dev)s",
             "udevadm info -a %(dev)s"
         ]
-        self.add_blockdev_cmd(cmds, blacklist='ram.*')
+        self.add_blockdev_cmd(cmds, blacklist=['ram.*', 'loop.*'])
+        loop_devices = self.exec_cmd('losetup --all --noheadings')
+        if loop_devices['status'] == 0:
+            loop_cmds = ["fdisk -l", "udevadm info", "udevadm info -a"]
```

Comment:
Why we cant use `cmds` list and set here `loop_cmds = cmds[1:]` (and add a warning comment to `cmds` declaration that `parted` command must be the first)?

This will prevent adding a command to two lists in future.

Reply:
Why are we re-detecting `loopX` devices here? Is there something wrong with the list determined and used by `add_blockdev_cmd()` normally?

I.E. why can't we simply do this?

```python
self.add_blockdev_cmd(cmds, blacklist=['ram.*', 'loop.*'])
self.add_blockdev_cmd(cmds[1:], whitelist='loop.*')
```

Reply:
+1 for this simplistic elegant proposal.

Reply:
It didn't occur to me to do it this way, and is super elegant. I'll make this change and test it, then commit again.

Reply:
OK I remember why I decided to gather the devices again. The problem is that when you detach an image from a loop device, the actual loop device continues existing in /dev. An example:

```
[root@voyager ~]# ll /dev/loop*
brw-rw----. 1 root disk   7,   0 Feb 14 14:42 /dev/loop0
brw-rw----. 1 root disk 259,   0 Feb 14 14:42 /dev/loop0p1
brw-rw----. 1 root disk   7,   1 Feb 12 10:00 /dev/loop1
brw-rw----. 1 root disk   7,   2 Feb 14 14:42 /dev/loop2
crw-rw----. 1 root disk  10, 237 Oct 22 15:49 /dev/loop-control

[root@voyager ~]# losetup -d /dev/loop1

[root@voyager ~]# losetup --all --noheadings
/dev/loop2: [64513]:2152 (/root/loopbackfile2.img)
/dev/loop0: [64513]:84 (/root/loopbackfile.img)

[root@voyager ~]# ll /dev/loop*
brw-rw----. 1 root disk   7,   0 Feb 14 14:42 /dev/loop0
brw-rw----. 1 root disk 259,   0 Feb 14 14:42 /dev/loop0p1
brw-rw----. 1 root disk   7,   1 Feb 14 14:43 /dev/loop1
brw-rw----. 1 root disk   7,   2 Feb 14 14:42 /dev/loop2
crw-rw----. 1 root disk  10, 237 Oct 22 15:49 /dev/loop-control
```

Result when running with the proposed change:

```
[root@voyager block]# cat fdisk_-l_.dev.loop1
fdisk: cannot open /dev/loop1: Inappropriate ioctl for device
```

As you can see, using 'losetup --all --noheadings' is the only way to have a proper list of loop devices in use.

Reply:
Hm.. should not we improve detection of (loop only?) devices in https://github.com/sosreport/sos/blob/ade857c82926bb7de2c45232f00a3f346db4e59a/sos/report/__init__.py#L490 , then? Can't this issue affect some other calls of `add_blockdev_cmd` against loop devices?

Reply:
Agreed @pmoravec

@jcastill - let's move this detection change into `sos/report/__init__.py.get_block_devs()`, then we can use the simplified calls above in the plugin while having better `loopX` device detection across all plugins.

Reply:
@TurboTurtle @pmoravec ack, I'll start work on this right away, then I'll resubmit this PR when the changes on the other one are added.

Reply:
My setup has loop device count up to 10K. The plugin times out while collecting data. What will be impact of this change? Is there an option to skip loop devices?

Reply:
@vikas-goel I'm rewriting this PR at the moment to deal with loop devices outside of the block plugin, like we do with block devices etc. While the PR is almost ready, I'm doing some tests before sending it, so I'll test with 10K loop devices and see if my new approach times out as well.